src/Microsoft.OData.Core/MultipartMixed/ODataMultipartMixedBatchWriter.cs

```diff
@@ -0,0 +1,413 @@
+        }
+
+#if PORTABLELIB
+        /// <summary>
+        /// Flush the output.
+        /// </summary>
+        /// <returns>Task representing the pending flush operation.</returns>
+        protected override Task FlushAsynchronously()
+        {
+            return this.RawOutputContext.FlushAsync();
+        }
+#endif
+
+        /// <summary>
+        /// Starts a new changeset - implementation of the actual functionality.
+        /// </summary>
+        /// <param name="groupId">Ignored value for multipart batch start changeset implementation
+        /// since in this format the changeset boundary is always created randomly for both request and response.</param>
+        protected override void WriteStartChangesetImplementation(/*ignored*/string groupId)
+        {
+            // write pending message data (headers, response line) for a previously unclosed message/request
+            this.WritePendingMessageData(true);
+
+            this.SetState(BatchWriterState.ChangesetStarted);
+            Debug.Assert(this.changeSetBoundary == null, "this.changeSetBoundary == null");
+            this.changeSetBoundary = ODataMultipartMixedBatchWriterUtils.CreateChangeSetBoundary(this.RawOutputContext.WritingResponse);
+
+            // write the boundary string
+            ODataMultipartMixedBatchWriterUtils.WriteStartBoundary(this.RawOutputContext.TextWriter, this.batchBoundary, !this.batchStartBoundaryWritten);
+            this.batchStartBoundaryWritten = true;
+
+            // write the change set headers
+            ODataMultipartMixedBatchWriterUtils.WriteChangeSetPreamble(this.RawOutputContext.TextWriter, this.changeSetBoundary);
+            this.changesetStartBoundaryWritten = false;
+        }
+
+        /// <summary>
+        /// Creates an <see cref="ODataBatchOperationRequestMessage"/> for writing an operation of a batch request
+        /// - implementation of the actual functionality.
+        /// </summary>
+        /// <param name="method">The Http method to be used for this request operation.</param>
+        /// <param name="uri">The Uri to be used for this request operation.</param>
+        /// <param name="contentId">The Content-ID value to write in ChangeSet head.</param>
+        /// <param name="payloadUriOption">
+        /// The format of operation Request-URI, which could be AbsoluteUri, AbsoluteResourcePathAndHost, or RelativeResourcePath.</param>
+        /// <param name="dependsOnIds">The prerequisite request ids of this request.
+        /// For multipart format, it is less significant than it is for Json batch format and has a default value of null.</param>
+        /// <returns>The message that can be used to write the request operation.</returns>
+        protected override ODataBatchOperationRequestMessage CreateOperationRequestMessageImplementation(
+            string method, Uri uri, string contentId, BatchPayloadUriOption payloadUriOption,
+            /*ignored*/IEnumerable<string> dependsOnIds)

```

Comment:
>/*ignored*/IEnumerable<string> dependsOnIds [](start = 12, length = 43)

Rather than ignore, should validate that the dependsOnIds are all contentIds that have previously been read.

Reply:
this will be related to introducing dependsOn to multipart batch side.

---
In reply to: [155388122](https://github.com/OData/odata.net/pull/992#discussion_r155388122) [](ancestors = 155388122)